internal/metamorphic/crossversion/crossversion_test: TestMetaCrossVersion failed

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/21323833?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/21323833?buildTab=artifacts#/) on master @ [7ba7d544ebaa](https://github.com/cockroachdb/pebble/commits/7ba7d544ebaa):


```
                --- FAIL: TestMeta/execution/random-018 (0.58s)
                --- FAIL: TestMeta/execution/random-014 (0.46s)
                --- FAIL: TestMeta/execution/random-022 (0.43s)
                --- FAIL: TestMeta/execution/random-005 (0.42s)
                --- FAIL: TestMeta/execution/random-013 (0.42s)
                --- FAIL: TestMeta/execution/random-024 (0.54s)
                --- FAIL: TestMeta/execution/random-012 (0.49s)
                --- FAIL: TestMeta/execution/random-011 (0.53s)
                --- FAIL: TestMeta/execution/random-010 (0.48s)
                --- FAIL: TestMeta/execution/random-008 (0.46s)
                --- FAIL: TestMeta/execution/random-007 (0.39s)
                --- FAIL: TestMeta/execution/random-006 (0.39s)
                --- FAIL: TestMeta/execution/random-003 (0.38s)
                --- FAIL: TestMeta/execution/random-004 (0.44s)
                --- FAIL: TestMeta/execution/random-001 (0.55s)
                --- FAIL: TestMeta/execution/standard-029 (0.35s)
                --- FAIL: TestMeta/execution/random-026 (0.48s)
                --- FAIL: TestMeta/execution/random-000 (0.44s)
                --- FAIL: TestMeta/execution/standard-028 (0.36s)
                --- FAIL: TestMeta/execution/random-002 (0.73s)
                --- FAIL: TestMeta/execution/standard-027 (0.42s)
                --- FAIL: TestMeta/execution/standard-013 (0.42s)
                --- FAIL: TestMeta/execution/standard-024 (0.39s)
                --- FAIL: TestMeta/execution/standard-025 (0.42s)
                --- FAIL: TestMeta/execution/standard-023 (0.36s)
                --- FAIL: TestMeta/execution/standard-022 (0.34s)
                --- FAIL: TestMeta/execution/standard-020 (0.31s)
                --- FAIL: TestMeta/execution/standard-019 (0.37s)
                --- FAIL: TestMeta/execution/standard-018 (0.36s)
                --- FAIL: TestMeta/execution/standard-021 (0.64s)
                --- FAIL: TestMeta/execution/standard-016 (0.40s)
                --- FAIL: TestMeta/execution/standard-015 (0.38s)
                --- FAIL: TestMeta/execution/standard-017 (0.52s)
                --- FAIL: TestMeta/execution/standard-014 (0.38s)
                --- FAIL: TestMeta/execution/standard-012 (0.42s)
                --- FAIL: TestMeta/execution/standard-007 (0.46s)
                --- FAIL: TestMeta/execution/standard-011 (0.33s)
                --- FAIL: TestMeta/execution/standard-010 (0.36s)
                --- FAIL: TestMeta/execution/standard-009 (0.38s)
                --- FAIL: TestMeta/execution/standard-008 (0.38s)
                --- FAIL: TestMeta/execution/standard-006 (0.39s)
                --- FAIL: TestMeta/execution/standard-002 (0.38s)
                --- FAIL: TestMeta/execution/standard-005 (0.40s)
                --- FAIL: TestMeta/execution/standard-003 (0.33s)
                --- FAIL: TestMeta/execution/standard-001 (0.31s)
                --- PASS: TestMeta/execution/standard-004 (1.20s)
                --- PASS: TestMeta/execution/random-009 (4.83s)
        FAIL
        
        --- FAIL: TestMetaCrossVersion/execution-master/sha=52cdf9e7-seed=7263098007406854491-opts=random-013(sha=1fa6f6e1-seed=1325344475349087077-opts=standard-018(sha=22899134-seed=11363647126493835201-opts=random-012(sha=9fab7356-seed=1146376589077569909-opts=random-005(sha=8b6bc41b-seed=11587628696746038297-opts=standard-012(<empty>))))) (33.49s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/test-eng
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: PEBBLE-1424

internal/metamorphic/crossversion/crossversion_test.TestMetaCrossVersion [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/21325341?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_NightlyPebbleMetamorphicCrossversion/21325341?buildTab=artifacts#/) on master @ [758587f91744](https://github.com/cockroachdb/pebble/commits/758587f91744):


```
                --- FAIL: TestMeta/execution/standard-019 (0.07s)
                --- FAIL: TestMeta/execution/standard-015 (0.20s)
                --- FAIL: TestMeta/execution/standard-016 (0.06s)
                --- FAIL: TestMeta/execution/standard-021 (0.14s)
                --- FAIL: TestMeta/execution/random-028 (0.05s)
                --- FAIL: TestMeta/execution/random-003 (0.07s)
                --- FAIL: TestMeta/execution/random-026 (0.20s)
                --- FAIL: TestMeta/execution/random-010 (0.07s)
                --- FAIL: TestMeta/execution/random-002 (0.16s)
                --- FAIL: TestMeta/execution/standard-008 (0.04s)
                --- FAIL: TestMeta/execution/standard-014 (0.04s)
                --- PASS: TestMeta/execution/standard-024 (0.40s)
                --- FAIL: TestMeta/execution/random-011 (0.08s)
                --- PASS: TestMeta/execution/random-000 (0.57s)
                --- FAIL: TestMeta/execution/standard-012 (0.06s)
                --- FAIL: TestMeta/execution/standard-011 (0.06s)
                --- FAIL: TestMeta/execution/standard-010 (0.06s)
                --- FAIL: TestMeta/execution/standard-007 (0.05s)
                --- PASS: TestMeta/execution/standard-018 (0.47s)
                --- FAIL: TestMeta/execution/standard-009 (0.10s)
                --- FAIL: TestMeta/execution/standard-006 (0.11s)
                --- PASS: TestMeta/execution/standard-017 (0.52s)
                --- FAIL: TestMeta/execution/random-024 (0.08s)
                --- FAIL: TestMeta/execution/random-005 (0.11s)
                --- FAIL: TestMeta/execution/random-025 (0.07s)
                --- FAIL: TestMeta/execution/random-021 (0.06s)
                --- FAIL: TestMeta/execution/random-019 (0.09s)
                --- FAIL: TestMeta/execution/standard-002 (0.06s)
                --- FAIL: TestMeta/execution/random-020 (0.09s)
                --- PASS: TestMeta/execution/standard-013 (0.35s)
                --- FAIL: TestMeta/execution/standard-003 (0.06s)
                --- FAIL: TestMeta/execution/random-009 (0.06s)
                --- FAIL: TestMeta/execution/random-014 (0.08s)
                --- FAIL: TestMeta/execution/random-018 (0.08s)
                --- FAIL: TestMeta/execution/random-017 (0.09s)
                --- FAIL: TestMeta/execution/random-016 (0.07s)
                --- PASS: TestMeta/execution/random-027 (0.69s)
                --- FAIL: TestMeta/execution/random-001 (0.10s)
                --- FAIL: TestMeta/execution/random-013 (0.13s)
                --- FAIL: TestMeta/execution/random-023 (0.06s)
                --- FAIL: TestMeta/execution/standard-001 (0.06s)
                --- PASS: TestMeta/execution/standard-004 (0.44s)
                --- PASS: TestMeta/execution/standard-005 (0.41s)
        FAIL
        
        --- FAIL: TestMetaCrossVersion/execution-master/sha=52cdf9e7-seed=2808616311506362991-opts=random-019(sha=1fa6f6e1-seed=14312934141372743513-opts=standard-011(sha=22899134-seed=13262527562117544579-opts=standard-024(sha=9fab7356-seed=9594665255541932392-opts=random-021(sha=8b6bc41b-seed=11589549659788405142-opts=standard-022(<empty>))))) (5.47s)
=== RUN   TestMetaCrossVersion/execution-master
    crossversion_test.go:278: Will run the metamorphic test 10 times, each time with a different initial state.
    crossversion_test.go:280:   Version: master (758587f9)
    --- FAIL: TestMetaCrossVersion/execution-master (54.76s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMetaCrossVersion.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/level_checker.go
+++ b/level_checker.go
@@ -731,14 +731,12 @@ func checkRangeKeyMetadata(c *checkConfig, allTables []*manifest.TableMetadata) 
 		if hasRangeKeySets && file.RangeKeyKinds != manifest.AnyRangeKeys {
 			return errors.Errorf("table %s has RangeKeyKinds!=AnyRangeKeys but contains RANGEKEYSETs", file.TableNum)
 		}
-		// For non-virtual tables, HasRangeKeys and RangeKeyKinds must exactly match reality.
 		if !file.Virtual {
+			// For non-virtual tables, HasRangeKeys must exactly match reality.
 			if file.HasRangeKeys && !hasRangeKeys {
 				return errors.Errorf("table %s has HasRangeKeys=true but contains no range keys", file.TableNum)
 			}
-			if !hasRangeKeySets && file.RangeKeyKinds == manifest.AnyRangeKeys {
-				return errors.Errorf("table %s has RangeKeyKinds=AnyRangeKeys but does not contain RANGEKEYSETs", file.TableNum)
-			}
+			// RangeKeyKinds allows for false positives even for non-virtual tables.
 		}
 	}
 	return nil

